prime-dotnet-webapi/Services/EnrolleeService.cs

```diff
@@ -495,6 +497,19 @@ private async Task CreateSelfDeclarationDocuments(int enrolleeId, ICollection<Se
                     });
                 }
             }
+
+            foreach (var document in currentSelfDeclarationDocuments)
+            {
+                if (!newDeclarations.Select(nd => nd.SelfDeclarationTypeCode).Contains(document.SelfDeclarationTypeCode))
+                {
+                    var documentToUpdate = await _context.SelfDeclarationDocuments
+                        .Where(sdd => sdd.EnrolleeId == enrolleeId && sdd.DocumentGuid == document.DocumentGuid)
+                        .SingleOrDefaultAsync();
+
+                    documentToUpdate.Hidden = true;
+                    await _context.SaveChangesAsync();
```

Comment:
You don't need to save on every document; in fact you should probably not be saving here at all in the middle of updating the Enrollee.